lib/puppet/util/windows/eventlog.rb

```diff
@@ -31,7 +31,7 @@ class Puppet::Util::Windows::EventLog
   def initialize(source_name = 'Puppet')
     @eventlog_handle = RegisterEventSourceW(FFI::Pointer::NULL, wide_string(source_name))
     if @eventlog_handle == NULL_HANDLE
-      raise EventlogError.new("failed to open Windows eventlog")
+      raise EventLogError.new("RegisterEventSourceW failed to open Windows eventlog (exit code #{FFI.errno})", FFI.errno)
```

Comment:
What if we were to just move the logic for the error generation to a method (instead of aliasing the type) - like

```ruby
def new_eventlog_error(msg, code = FFI.errno)
  defined?(Puppet::Util::Windows::Error) ?
    Puppet::Util::Windows::Error.new(msg) :
    SystemCallError.new(msg + " (Win32 error: #{code})")
end
```

And then the callers can just do this:

```ruby
raise new_eventlog_error('something bad happened')
```